msg.go

```diff
@@ -619,23 +619,33 @@ func intToBytes(i *big.Int, length int) []byte {
 // PackRR packs a resource record rr into msg[off:].
 // See PackDomainName for documentation about the compression.
 func PackRR(rr RR, msg []byte, off int, compression map[string]int, compress bool) (off1 int, err error) {
-	return packRR(rr, msg, off, compressionMap{ext: compression}, compress)
+	headerEnd, off1, err := packRR(rr, msg, off, compressionMap{ext: compression}, compress)
+	if err == nil {
```

Comment:
doesn't this leave the same race?

Reply:
No it doesn’t, because `PackRR` isn’t called by the packing code which calls the internal `packRR`. This can be removed of course, but I can imagine someone might be relying on it.